data_processors/pipeline/domain/workflow.py

```diff
@@ -416,3 +416,19 @@ def must_have_acceptable_coverage_yield(self):
     def must_be_valid_for_analysis(self):
         # TODO
         return self
+
+
+class ICAResourceOverridesStep:
+    def __init__(self, step_id, resource_type, resource_size):
+        self.step_id = step_id
+        self.resource_type = resource_type
+        self.resource_size = resource_size
```

Comment:
Type and size are both limited/specific values, right? 
Would it make sense to have them in an enum rather than free text?
(perhaps even check that the type/size combination is a supported one)

Reply:
class enums for ICAResourceType and ICAResourceSize added in 1b5db9c3a3bfb91fbf6f9f5f116d1eed3fbe04b7